### ISSUE TITLE
EurekaClient without Registration

### ISSUE THREAD

**Issue by gondor** [spring-cloud/spring-cloud-netflix#1347]
One of the most common problems we've had in my organization in development is having to bring up a local Eureka and any related services.   

I've tried creating a configuration so engineers can turn on a discovery profile in development and it will use the remote Spring Cloud Netflix Eureka server as a lookup and no registration.

I don't want local dev instances registering as a source in the registry but want the ability they can use existing services in CI.

Is this possible?  Here's the config below I've been trying and I end up getting the following error:

```
2016-09-15 15:39:27,546 [PollingServerListUpdater-0] WARN  c.n.n.l.DiscoveryEnabledNIWSServerList "cid":""  - EurekaClient has not been initialized yet, returning an empty list
```

Config Profile:
```
---
spring:
  profiles: discovery

eureka:
  client:
    registerWithEureka: false
    fetchRegistry: true
    serviceUrl:
          defaultZone: http://remotehost:8761/eureka/
```

**Comment by ryanjbaxter** [comment]
When I use this config, I do not see the error.  Instead the client registers with Eureka but registers with a "DOWN" status.  Can you provide a sample project that reproduces the issue?

**Comment by spencergibb** [comment]
Also, it might behave differently with different versions. What version are you using?  https://github.com/spring-cloud/spring-cloud-commons/issues/9 will help with this.

**Comment by gondor** [comment]
Hi everyone, its based off of BRIXTON.SR5 running Spring boot 1.4.0.   As far as spring-cloud-commons its release is 1.1.1

**Comment by gondor** [comment]
Thanks everyone. I figured it out.  I was testing by adding an additional profile in a local ```application.yml``` as shown above.  All our services use a remote configuration server and I didn't realize the global value it was serving was disabling discovery (default in global profile).  I didn't realize that trumps my local value.

I added the above profile to our global configuration server and now its working perfectly.  Thank you.
